File: analytics/src/test/java/io/appmetrica/analytics/impl/DataSendingRestrictionControllerStorageImplTest.java
package io.appmetrica.analytics.impl;

import io.appmetrica.analytics.impl.db.preferences.PreferencesServiceDbStorage;
import io.appmetrica.analytics.impl.db.preferences.PreferencesServiceDbStorageTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.doReturn;
import static org.mockito.Mockito.verify;

public class DataSendingRestrictionControllerStorageImplTest {

    private final PreferencesServiceDbStorage mDbStorage = PreferencesServiceDbStorageTest.createMock();

    private DataSendingRestrictionControllerImpl.StorageImpl mStorage = new DataSendingRestrictionControllerImpl.StorageImpl(
        mDbStorage
    );

    @Test
    public void testStore() {
        mStorage.storeRestrictionFromMainReporter(true);
        verify(mDbStorage).putDataSendingRestrictedFromMainReporter(true);
        verify(mDbStorage).commit();
    }

    @Test
    public void testRead() {
        doReturn(false).when(mDbStorage).getDataSendingRestrictedFromMainReporter();
        assertThat(mStorage.readRestrictionFromMainReporter()).isFalse();
        verify(mDbStorage).getDataSendingRestrictedFromMainReporter();
    }
}

File: analytics/src/test/java/io/appmetrica/analytics/impl/DataSendingRestrictionControllerStorageImplTest.kt
package io.appmetrica.analytics.impl

import io.appmetrica.analytics.impl.DataSendingRestrictionControllerImpl.StorageImpl
import io.appmetrica.analytics.impl.db.preferences.PreferencesServiceDbStorage
import io.appmetrica.gradle.testutils.CommonTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.mockito.kotlin.doReturn
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify

internal class DataSendingRestrictionControllerStorageImplTest : CommonTest() {
    private val dbStorage: PreferencesServiceDbStorage = mock {
        on { dataSendingRestrictedFromMainReporter } doReturn false
    }

    private val storage by setUp { StorageImpl(dbStorage) }

    @Test
    fun `store restriction from main reporter`() {
        storage.storeRestrictionFromMainReporter(true)
        verify(dbStorage).putDataSendingRestrictedFromMainReporter(true)
    }

    @Test
    fun `read restriction from main reporter`() {
        assertThat(storage.readRestrictionFromMainReporter()).isFalse()
        verify(dbStorage).dataSendingRestrictedFromMainReporter
    }
}
